EKF/estimator_interface.cpp

```diff
@@ -404,19 +404,31 @@ void EstimatorInterface::setOpticalFlowData(uint64_t time_usec, flow_message *fl
 
 			// NOTE: the EKF uses the reverse sign convention to the flow sensor. EKF assumes positive LOS rate is produced by a RH rotation of the image about the sensor axis.
 			// copy the optical and gyro measured delta angles
+			imuSample matching_imu_sample;
+			Vector3f matching_gyro_sample;
+
+			/* for optical flow boards with no gyro. The driver will set the gyro x, y and z fields in the optical_flow message to NAN so that  
+			 * we can check this here and get the gyro data with the matching timestamp from the imu buffer. The gyro data is divided by dt to 
+			 * convert it into a rate before removing it from the flow to remove the rotation component. This is done to prevent errors when the flow
+			 * and gyro data are integrated over different times */
+			if (!ISFINITE(flow->gyrodata(0)) && !ISFINITE(flow->gyrodata(1)) && !ISFINITE(flow->gyrodata(2))) {
+				_imu_buffer.read_first_older_than(optflow_sample_new.time_us, &matching_imu_sample);
+				flow->gyrodata = matching_imu_sample.delta_ang / matching_imu_sample.delta_ang_dt * delta_time;
```

Comment:
This introduces point sampling of gyro data which is going to be problematic with lower frequency optical flow data, eg 10Hz due to aliasing of body motion. A different approach is going to be needed. We already have _imu_del_ang_of and _delta_time_of class variables which are the accumulated gyro data between the current and previous optical flow observation. I think we should move the motion compensation into controlOpticalFlowFusion() and use those variables